extensions/parquet/table/src/main/java/io/deephaven/parquet/table/transfer/PrimitiveVectorTransfer.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2016-2023 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.parquet.table.transfer;
+
+import io.deephaven.engine.rowset.RowSequence;
+import io.deephaven.engine.table.ColumnSource;
+import io.deephaven.vector.Vector;
+import org.jetbrains.annotations.NotNull;
+
+import java.nio.Buffer;
+
+/**
+ * Used as a base class of transfer objects for vectors of primitive types.
+ */
+abstract class PrimitiveVectorTransfer<T extends Vector<?>, B extends Buffer>
```

Comment:
Why don't you have a similar `PrimitiveArrayTransfer` class? I would either remove this layer, or have the same layering in the array case, unless I'm missing some complexity there.